### PR TITLE
Replaces `move_on_shuttle` with a trait

### PR DESCRIPTION
**Pull request by san7890**
## About The Pull Request

`move_on_shuttle` was a variable that existed on `/mob` despite only ever being used by cameras. This PR downgrades the variable to only exist on `/mob/camera` types, but have the only thing that variable do is add a trait that blocks movement on shuttles if not allowed. This allows us to check the variable without casting to camera, which is what some code was un-necessarily doing. This also retainst the ability to add the trait later on to different mobs who we may not want to be on shuttles that aren't `/mob/camera`.
## Why It's Good For The Game

There's no point for this variable to live on `/mob` if all it can do is clog up the VV screen, especially if the current utilization for it is only cameras. Let's move it off `/mob` and turn it into a trait so it still has the similar utilization on a global level without typecasting - as well as make it actually work since needless typecasting was introducing a minor bug
## Changelog
Irrelevant.

### CODE REVIEW

**Review by Time-Green — CHANGES_REQUESTED** [review]

File: code/modules/shuttle/on_move.dm [hunk 1]
```diff
@@ -264,12 +264,12 @@ All ShuttleMove procs go here
...
 
 /mob/onShuttleMove(turf/newT, turf/oldT, list/movement_force, move_dir, obj/docking_port/stationary/old_dock, obj/docking_port/mobile/moving_dock)
-	if(!move_on_shuttle)
+	if(!HAS_TRAIT(src, TRAIT_BLOCK_SHUTTLE_MOVEMENT))
 		return
 	. = ..()
 
 /mob/afterShuttleMove(turf/oldT, list/movement_force, shuttle_dir, shuttle_preferred_direction, move_dir, rotation)
-	if(!move_on_shuttle)
+	if(!HAS_TRAIT(src, TRAIT_BLOCK_SHUTTLE_MOVEMENT))
```
[the remark this answers is not in the document]
Comment (Time-Green) [reply code/modules/shuttle/on_move.dm:272]: Im tired but these look inverted

[the remark this answers is not in the document]
Comment (san7890) [reply code/modules/shuttle/on_move.dm:272]: you're right whoops

File: code/modules/shuttle/arrivals.dm [hunk 2]
```diff
@@ -112,14 +112,9 @@
...
 /obj/docking_port/mobile/arrivals/proc/PersonCheck()
-	for(var/V in GLOB.player_list)
-		var/mob/M = V
-		if((get_area(M) in areas) && M.stat != DEAD)
-			if(!iscameramob(M))
-				return TRUE
-			var/mob/camera/C = M
-			if(C.move_on_shuttle)
-				return TRUE
+	for(var/mob/player in GLOB.player_list)
```
[the remark this answers is not in the document]
Comment (san7890) [reply code/modules/shuttle/arrivals.dm:115]: i was scared of potentially qdeleted mobs but i think destroy handles this list well enough so sure

[the remark this answers is not in the document]
Comment (Time-Green) [reply code/modules/shuttle/arrivals.dm]: It didn’t check before so it’s probably fine?

**Pull request merged**